SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -0,0 +1,547 @@
+        su2double artcompc2 = SolT * dRhodT * scalar / (Density);
+
+        LinSysRes[total_index] += artcompc1 + artcompc2;
+
+        /*--- Add the extra Jacobian term to the scalar system. ---*/
+
+        su2double Jaccomp = scalar * dRhodC + Density;
+        su2double JacTerm = Jaccomp * Delta;
+
+        Jacobian.AddVal2Diag(iPoint, JacTerm);
+      }
+    }
+  }
+  END_SU2_OMP_FOR
+}
+
+void CSpeciesFlameletSolver::Source_Residual(CGeometry* geometry, CSolver** solver_container,
+                                             CNumerics** numerics_container, CConfig* config, unsigned short iMesh) {
+  const bool implicit = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
+
+  auto* flowNodes = su2staticcast_p<CFlowVariable*>(solver_container[FLOW_SOL]->GetNodes());
+  auto* first_numerics = numerics_container[SOURCE_FIRST_TERM + omp_get_thread_num() * MAX_TERMS];
+
+  SU2_OMP_FOR_DYN(omp_chunk_size)
+  for (auto i_point = 0u; i_point < nPointDomain; i_point++) {
+    /*--- Set primitive variables w/o reconstruction. ---*/
+
+    first_numerics->SetPrimitive(flowNodes->GetPrimitive(i_point), nullptr);
+
+    /*--- Set scalar variables w/o reconstruction. ---*/
+
+    first_numerics->SetScalarVar(nodes->GetSolution(i_point), nullptr);
+
+    first_numerics->SetDiffusionCoeff(nodes->GetDiffusivity(i_point), nodes->GetDiffusivity(i_point));
+
+    /*--- Set volume of the dual cell. ---*/
+
+    first_numerics->SetVolume(geometry->nodes->GetVolume(i_point));
+
+    /*--- Retrieve scalar sources from CVariable class and update numerics class data. ---*/
+    first_numerics->SetScalarSources(nodes->GetScalarSources(i_point));
+
+    auto residual = first_numerics->ComputeResidual(config);
+
+    /*--- Add Residual. ---*/
+
+    LinSysRes.SubtractBlock(i_point, residual);
+
+    /*--- Implicit part. ---*/
+
+    if (implicit) Jacobian.SubtractBlock2Diag(i_point, residual.jacobian_i);
+  }
+  END_SU2_OMP_FOR
+
+  /*--- call the species solver for the shared sources (axisymmetric) ---*/
+  CSpeciesSolver::Source_Residual(geometry, solver_container, numerics_container, config, iMesh);
+}
+
+void CSpeciesFlameletSolver::BC_Inlet(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                                      CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) {
+  string Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+
+  su2double temp_inlet = config->GetInlet_Ttotal(Marker_Tag);
+  const su2double* scalar_inlet = config->GetInlet_SpeciesVal(Marker_Tag);
+
+  /*--- We compute inlet enthalpy from the temperature and progress variable. ---*/
+  su2double enth_inlet = scalar_inlet[I_ENTH];
+  solver_container[FLOW_SOL]->GetFluidModel()->GetEnthFromTemp(enth_inlet,
+                scalar_inlet[I_PROGVAR], temp_inlet, scalar_inlet[I_ENTH]);
+ 
+  for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
```

Comment:
Add SU2_OMP_FOR_STAT as done in other BC please

Reply:
Correct, because chunk_size is computed based on `nPoint` which is much larger than the number of vertices in a marker.